react/constants/SearchHelpers.js

```diff
@@ -0,0 +1,58 @@
+import QueryString from 'query-string'
+
+/**
+ * Returns a string representing the facets param based in the location pathName and queryString.
+ * e.g: smartphones?map=c
+ * e.g2: eletronics/smartphones?map=c,c
+ */
+export function getFacetsFromURL(pathName, queryParams, isBrand) {
+  const { query, map } = getSearchParamsFromURL(pathName, queryParams, isBrand)
+  return `${query}${map ? `?map=${map}` : ''}`
+}
+
+export function getSearchParamsFromURL(pathName, queryParams, isBrand) {
+  let map = ''
```

Comment:
You can empower this entire file with ramda functions.

Reply:
how?